apps/server/src/modules/files-storage/client/s3-client.adapter.ts

```diff
@@ -189,37 +187,43 @@ export class S3ClientAdapter implements IStorageClient {
 		return this.client.send(req);
 	}
 
-	public async list(prefix: string, maxKeys?: number) {
-		this.logger.log({ action: 'list', params: { prefix, bucket: this.config.bucket } });
-
+	public async list(params: IListFiles) {
 		try {
-			let files: string[] = [];
-			let ret: ListObjectsV2CommandOutput | undefined;
+			this.logger.log({ action: 'list', params });
 
-			do {
-				const req = new ListObjectsV2Command({
-					Bucket: this.config.bucket,
-					Prefix: prefix,
-					ContinuationToken: ret?.NextContinuationToken,
-					MaxKeys: maxKeys && maxKeys - files.length,
-				});
+			const result = await this.listObjectKeysRecursive(params);
+
+			return result;
+		} catch (err) {
+			throw new InternalServerErrorException('S3ClientAdapter:listDirectory');
+		}
+	}
```

Comment:
I'd prefer the interface to be as simple as possible, that's why I advocated for the split into internal and external
This way the tests could also stay the same.

```typescript
	public async list(path: string, maxKeys?: number) {
		try {
			this.logger.log({ action: 'list', params });

			const { files } = await this.listObjectKeysRecursive({ path, maxKeys });

			// Because of your design of the recursive function, files may be undefined.
			// Should never happen in practice, but typescript believes it
			return files ?? [];
		} catch (err) {
			throw new InternalServerErrorException('S3ClientAdapter:listDirectory');
		}
	}
```

Reply:
No, files are not undefined
let files: string[] = params.files ? params.files : [];

And no, we leave the interface open. Maybe later we refactor this

Reply:
You're right, I thought listObjectKeysRecursive returns the same IListFiles Interface. Should have looked at the inferred types.

I don't see the point of splitting it into a public/private function if both functions have the exact same interface, but if you want to leave it open I'll accept that